plugins/org.jkiss.dbeaver.ext.dm/plugin.xml

```diff
@@ -0,0 +1,243 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?eclipse version="3.2"?>
+<!--
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+  -->
+
+<plugin>
+    <extension point="org.jkiss.dbeaver.dataSourceProvider">
+        <datasource
+                class="org.jkiss.dbeaver.ext.dm.DMDataSourceProvider"
+                description="DM connector"
+                id="dm"
+                parent="generic"
+                label="DaMeng"
+                icon="icons/dm_icon.png"
+                dialect="dm">
+            <tree path="dm"
```

Comment:
As I can see, this tree does not have many special classes and cases. It is again a copy-paste code.
So, please use treeInjection for extra classes instead, like in the Oceanbase code:

https://github.com/dbeaver/dbeaver/blob/359a137b918c29ab5953493e772ced451c97749b/plugins/org.jkiss.dbeaver.ext.oceanbase/plugin.xml#L29-L32

Reply:
Dameng is different from generic, it is similar to Oracle. Can these be retained to support other features?

Reply:
Another example of generic:

https://github.com/dbeaver/dbeaver/blob/653cab6228bd90f4c664bcf959e561f1f0f5d831/plugins/org.jkiss.dbeaver.ext.h2/plugin.xml#L19-L24

And you can try to do it like Oracle-based (you can look at the Oceanbase, based on MySQL parent). But it will be much more complex because we have so many Oracle-specific code, queries, and other staff, which are not dependent on the driver. And for generics - more work is on the driver side (reading metadata, as an example).